core/src/main/java/org/eclipse/hono/util/CredentialsObject.java

```diff
@@ -0,0 +1,81 @@
+/**
+ * Copyright (c) 2017 Bosch Software Innovations GmbH.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *    Bosch Software Innovations GmbH - initial creation
+ */
+package org.eclipse.hono.util;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Encapsulates the credentials information for a device that was found by the get operation of the
+ * <a href="https://www.eclipse.org/hono/api/Credentials-API/">Credentials API</a>.
+ * <p>
+ * Is mapped internally from json representation by jackson-databind.
+ */
+public final class CredentialsObject {
+    @JsonProperty(CredentialsConstants.FIELD_DEVICE_ID)
+    private String deviceId;
+    @JsonProperty(CredentialsConstants.FIELD_TYPE)
+    private String type;
+    @JsonProperty(CredentialsConstants.FIELD_AUTH_ID)
+    private String authId;
+    @JsonProperty(CredentialsConstants.FIELD_ENABLED)
+    private Boolean enabled;
+    /*
+     * Since the format of the secrets field is not determined by the Credentials API, they are best represented as
+     * key-value maps with key and value both of type String.
+     * The further processing of secrets is part of the validator for the specific type.
+     */
+    @JsonProperty(CredentialsConstants.FIELD_SECRETS)
+    private List<Map<String,String>> secrets;
```

Comment:
That is fine, I will work on providing concrete classes after we made the M8 milestone directly :+1:.

Regarding the JsonObjects: I am sure we can do it, but as far as I see it only by providing an own jackson deserializer (which I wanted to avoid so far). Possibly this will have to be done for providing the concrete secrets objects anyway, so I would made another attempt for this then as well.
I understand well that you rather like to only have one kind of "container api" in Hono in contrast to mainly have the vertx Json-API and for this part the Java List- and Map-API.
Will come back to this later then.